snuba/datasets/configuration/json_schema.py

```diff
@@ -444,6 +482,15 @@ def registered_class_schema(
             "type": ["string", "null"],
             "description": "The column name, if this entity is partitioned, to select slice",
         },
+        "entity_subscription": {
+            "type": "object",
+            "description": "Specifies the processors and validators associated with subscriptions",
+            "properties": {
+                "processors": ENTITY_SUBSCRIPTION_PROCESSORS,
+                "validators": ENTITY_SUBSCRIPTION_VALIDATORS,
```

Comment:
This pattern of separate processors and validators seems like a pretty significant departure from what was discussed offline (having linked processors/validators so that a single class would both validate the presence of the org ID then include it into the query). Reasons for the change?

Reply:
I think there might have be some slight confusion with the selection of splitting or combining `processors` and `validators` offline. The problem with combining `processors` and `validators` into a single "processor" class is because currently, the `EntitySubscription` validation and query processing are two completely separate things. We currently do not validate org ID. The only validation we do right now is on the aggregation clauses of the query. On the contrary, we validate the aggregation but this does not mean we do some processing on the aggregation portion of the query. As a result, in the current state of the world, these two are separate components which live separately.

Reply:
I think there might be some slight confusing because I thought the option of splitting `processors` and `validators` was selected offline. The problem with combining processors and validators into a single processor class is because in the current state we run `EntitySubscriptions`, validation and query processing are two completely separate things. We currently do not validate org ID. We validate that the aggregation clauses of the query follow some specification in the query. On the contrary, we validate the aggregation but this does not mean we do some processing on the aggregation portion of the query. As a result, in the current state of the world, these two are separate components which live separately.

Reply:
If they are completely separate components why do they overlap in the `to_dict()` method. The fact that the validator can provide keys to be written to redis via `to_dict()` feels like an anti-pattern. It seems like it might be doing a lot more than validating.

Also, although we are not currently validating the org ID, shouldn't we be? That feels like it might be an oversight

Reply:
Forgot to comment here directly, but the PR has been updated and `to_dict` abstract method has been removed from the `EntitySubscriptionValidator` class.

I think that's a good point and we should be validating the org ID. Since the current state of entity subscriptions does not do org_id validation either and adding a new validator seems like more code that can complicate this PR, I can file a ticket for this feature?

Reply:
Follow up seems fine to me. I was not really suggesting we should be adding a new validator in this PR though. It's about whether the validator and processor should be a shared implementation as in this case of org_id they will be inherently linked. In this PR, the validator is completely separate so user need to remember to include it rather than it being an extension of the processor.

Reply:
That makes sense, I agree that in the case of `org_id`, the validator and processor is inherently linked and shared implementation would be a good safe-guard. Something I just thought of, for any processor a user defines, shouldn't the associated validator be automatically applied as opposed needing them to specific it explicitly? Could be out of scope.

Could I suggest a more incremental approach to this? The reason why validators and processors are implemented separately is because it matches what we have today. I'd be happy to create a ticket for this design change and carry it forward. Making this change requires some refactoring of the `build_request()` method in [SubscriptionData](https://github.com/getsentry/snuba/blob/33bb2c04aa08c123b15a4b5831cce0ca3e898b1b/snuba/subscriptions/data.py#L145).